internal/nexodus/wg_deploy.go

```diff
@@ -33,13 +33,17 @@ func (ax *Nexodus) DeployWireguardConfig(updatedPeers map[string]public.ModelsDe
 		}
 		// add routes for each peer candidate (unless the key matches the local nodes key)
 		peer, ok := cfg.Peers[updatedPeer.PublicKey]
-		if !ok || peer.PublicKey == ax.wireguardPubKey {
+		if !ok || peer.PublicKey == nx.wireguardPubKey {
 			continue
 		}
-		ax.handlePeerRoute(peer)
-		ax.handlePeerTunnel(peer)
+		if err := nx.handlePeerRoute(peer); err != nil {
+			return err
+		}
+		if err := nx.handlePeerTunnel(peer); err != nil {
+			return err
+		}
```

Comment:
I think a more optimistic way to handle this is to finish looping through all `updatedPeers` even if we encounter a failure with one of them. Configuring other peers may still work. You could just save off the first error you hit to return after the loop.